packages/api-explorer/src/components/SelectorContainer/SdkLanguageSelector.tsx

```diff
@@ -55,6 +57,20 @@ export const SdkLanguageSelector: FC = () => {
     setSdkLanguageAction({ sdkLanguage: language })
   }
 
+  useEffect(() => {
+    const searchParams = new URLSearchParams(history.location.search)
+    const urlSdk = searchParams.get('sdk')
+    if (urlSdk && allSdkLanguages.find((lang) => lang.label === urlSdk)) {
+      setSdkLanguageAction({ sdkLanguage: urlSdk! })
+    }
+  }, [])
+
+  useEffect(() => {
+    navigate(history.location.pathname, {
+      sdk: selectedSdkLanguage.toLowerCase(),
+    })
```

Comment:
based on the implementation we discussed for navigate I would expect this call to be along the lines of 
```
navigate(history.location.pathname, {...existingParams, sdk: selectSdkLanguage.toLowerCase()}
```

Reply:
If we just pass in `sdk: sdk: selectSdkLanguage.toLowerCase()`, the remaining params (or the `existingParams` which you note) would be undefined, so they would remain in the URL. It would be redundant to have existing params then wouldn't it?

Reply:
agreed. I wrote this before I saw and replied to your comment [here](https://github.com/looker-open-source/sdk-codegen/pull/1123#discussion_r930358895)